src/main/java/core/basesyntax/strategy/operations/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exceptions.NotEnoughFruitException;
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handleOperation(FruitTransaction fruitTransaction) {
+        ifPresent(fruitTransaction.getFruit(), fruitTransaction.getQuantity());
+    }
+
+    private void ifPresent(String name, int quantity) {

```

Comment:
Name `ifPresent` means that you need to check on existing but not means that the program will add some data anywhere. Rename to `processIfPresent` or sth like this